### PR TITLE
Add interactive functions to mark completed task and delete completed tasks

### PR DESCRIPTION
**Pull request by lenkon**
Project name: To Do list - interactive list

Project description: In this project, some functionality to the application will be added to make it interactive. The user will also be able to mark task completion by selecting the corresponding checkbox (or undo it by un-checking the checkbox). The updated tasks list will be stored in local storage.

#### Project requirements:
- Add a new JavaScript file and import it as a module:    
    - it will contain methods related to the status updates (`completed`: `true` / `false`).
- Add event listener to the checkbox (`change`).
- Update items object's value for `completed` key upon user actions.
- Implement a function for the "Clear all completed" button (use `filter()` method).
- Store the updated array of items in local storage, so the user gets the correct list of values after the page reloads (which means that any changes made to the list should be preserved).

Project requirements link: https://github.com/microverseinc/curriculum-javascript/blob/main/todo-list/m2_interactive_list.md

### CODE REVIEW

File: src/modules/toDoList.js [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+  }
+
+  saveTaskInList = (description, completed = false) => {
+    this.tasks.push({ description, completed, index: this.getTasks().length + 1 });
+    this.saveToStorage();
+  }
+
+  deleteTaskFromList = (clear = false, itemIndex = -1) => {
+    if (clear) {
+      this.tasks = this.getTasks().filter((task) => task.completed === false);
```
Comment (ichala) [inline src/modules/toDoList.js:43]: - [ ] while everything works as it supposes to be, there's a minor issue (i think its a visual issue) when I mark a task as completed, it does change to true, but when I refresh, even so, it's true, the check box is not displayed as checked, kindly fix that 🙏 

_potential solution: add a check when generating the task if it's completed then you show the check box as checked_

scroll

**Review by ichala — CHANGES_REQUESTED** [review]
Hi Student
Good job so far! 
👏👏👏 
You still need to work on some issues to go to the next project, but you're almost there!

the-simpsons-homer-simpson

## To Highlight 🎉

- No Linter errors  ✔️ 
- All functionalities implemented ✔️


 ## Required Changes ♻️


- [ ] **_🛑  Check The Comments Below_**





## Optional suggestions

- N/A

**********
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/modules/toDoList.js (hunk 1 above)
Comment (lenkon) [reply src/modules/toDoList.js:42]: Hi @ichala, Thanks for the valuable feedback. I have made the changes in the checkbox functionality according to the suggestions. Now, it's working fine.

**Review by AmaduKamara — APPROVED** [review]
# Status: Approved 🟢 🟢 🟢

Hi @lenkon,

Good job working on the changes asked by the previous reviewer 👍🏽

Your project is complete!  There is nothing else to say other than... it's time to merge it: ship it:
Congratulations! 🎉 

## Highlights:
✅ You used Gitflow correctly 👍🏽 
✅ Linter checks passed 👍🏽
✅ Todo list is interactive as expected 👍🏽
✅ Good communication with the previous reviewer 👍🏽

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Pull request merged**